helm/flowforge/templates/file-storage.yml

```diff
@@ -77,13 +77,58 @@ spec:
           capabilities:
             drop:
               - ALL
+      {{ if .Values.forge.localPostgresql -}}
+      - name: wait-for-local-db
+        image: "postgres:14"
+        imagePullPolicy: Always
+        command: ['sh', '-c', 'until pg_isready -h {{ include "forge.databaseHost" . }} -U {{ .Values.postgresql.auth.username }} -d {{ .Values.postgresql.auth.database }}; do sleep 2; done;']
+        securityContext:
+          readOnlyRootFilesystem: true
+          allowPrivilegeEscalation: false
+          capabilities:
+            drop:
+              - ALL
+        env:
+          - name: PGPASSWORD
+            valueFrom:
+              secretKeyRef:
+                name: {{ include "forge.secretName" . }}
+                key: postgres-password
+      - name: add-context-db
+        image: {{ .Values.postgresql.image.registry }}/{{ .Values.postgresql.image.repository }}:{{ .Values.postgresql.image.tag }}
+        command:
+        - /bin/sh
+        - /usr/local/upgrade.sh
```

Comment:
My only question on this is will it continue if this returns an error (e.g. because the db has already been created)?

Reply:
I've tested this script beforehand and file service restarts without any issues. It's because:

```
SELECT datname FROM pg_database WHERE datistemplate = false;
``` 
It's a simple select statement.
```
SELECT 'CREATE DATABASE "ff-context"' WHERE NOT EXISTS (SELECT FROM pg_database WHERE datname = 'ff-context')\gexec
```
Database creation is wrapped around a SELECT statement which checks if the database should be created.
```
GRANT ALL PRIVILEGES ON DATABASE "ff-context" TO "forge";
```
This statement does not fail, despite the existence of such permissions.

Overall, multiple script executions against the same database instance won't cause script failure, and as a result, the file service will be able to start properly.

Reply:
Thanks